src/internal.rs

```diff
@@ -1062,18 +1066,31 @@ pub struct Rav1dTileState {
     // each entry is one tile-sbrow; middle index is refidx
     pub lowest_pixel: usize,
 
-    pub dqmem: [[[u16; 2]; 3]; RAV1D_MAX_SEGMENTS as usize], /* [RAV1D_MAX_SEGMENTS][3 plane][2 dc/ac] */
-    pub dq: TileStateRef,
-    pub last_qidx: u8,
-    pub last_delta_lf: [i8; 4],
-    pub lflvlmem: [[[[u8; 2]; 8]; 4]; 8], /* [8 seg_id][4 dir][8 ref][2 is_gmv] */
-    pub lflvl: TileStateRef,
+    pub dqmem: RwLock<[[[u16; 2]; 3]; RAV1D_MAX_SEGMENTS as usize]>, /* [RAV1D_MAX_SEGMENTS][3 plane][2 dc/ac] */
+    pub dq: Atomic<TileStateRef>,
+    pub last_qidx: AtomicU8,
+    pub last_delta_lf: [AtomicI8; 4],
```

Comment:
See #1040 for how this works; it's quite simple now that `atomig` supports it.

Reply:
Ah yeah, now that atomig supports that it's a much better approach.

Reply:
Oh, there was one place where we were setting one element at a time instead of loading the entire array at once, but that was easy enough to rewrite into a single load and store.